crates/ruff_linter/src/rules/flake8_pyi/rules/collections_named_tuple.rs

```diff
@@ -13,16 +13,17 @@ use crate::checkers::ast::Checker;
 /// ## Why is this bad?
 /// `typing.NamedTuple` is the "typed version" of `collections.namedtuple`.
 ///
-/// The class generated by subclassing `typing.NamedTuple` is equivalent to
-/// `collections.namedtuple`, with the exception that `typing.NamedTuple`
-/// includes an `__annotations__` attribute, which allows type checkers to
-/// infer the types of the fields.
+/// Inheriting from `typing.NamedTuple` creates a custom `tuple` subclass in
+/// the same way as using the `collections.namedtuple` factory function.
+/// However, using `typing.NamedTuple` allows you to provide a type annotation
+/// for each field in the class. This means that type checkers will have much
+/// more information to work with, and will be able to analyze your code far
+/// more precisely.
```

Comment:
```suggestion
/// However, using `typing.NamedTuple` allows you to provide a type annotation
/// for each field in the class. This means that type checkers will have
/// more information to work with, and will be able to analyze your code
/// more precisely.
```

Maybe we can rephrase it to say that `namedtuple` types all fields as `Any` (I guess?) and that `typing.NamedTuple` allows you to specify the field types instead.

Reply:
> Maybe we can rephrase it to say that `namedtuple` types all fields as `Any` (I guess?)

That's correct, but I wonder if we really need to go into that much detail here... I think simply saying that it will make type-checking more accurate is okay for our purposes here, and helps avoid too much information overload